### PR TITLE
[AVCe] fixes for LA modes, maxFrameSize adding

### PR DESCRIPTION
**Pull request by ogolikov**

### CODE REVIEW

File: _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp [hunk 1]
```diff
@@ -1509,12 +1509,29 @@ mfxU8 LookAheadBrc2::GetQp(const BRCFrameParams& par)
 
     return mfxU8(m_curQp);
 }
+mfxU8 GetNewQP(mfxU32 size, mfxU32 targeSize, mfxU8 curQP)
+{
+    mfxF64 qstep = 0, qstep_new = 0;
+    qstep = QSTEP[IPP_MIN(51, curQP)];
+    qstep_new = qstep * pow((mfxF64)size / targeSize, 0.8);
+    mfxU8 qp_new = QStep2QpCeil(qstep_new);
+    if (qp_new < 51 && qstep_new >(QSTEP[qp_new] + QSTEP[qp_new + 1]) / 2)
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp]: this function is overcomplicated, I suggest something like following:
```
mfxU8 GetNewQP(mfxU32 size, mfxU32 targetSize, mfxU8 curQP)
{
    mfxF64 qstep     = QSTEP[std::min(51, curQP)];
    mfxF64 qstep_new = qstep * pow(mfxF64(size) / targetSize, 0.8);

    return QStep2QpRound(qstep_new);
}

mfxU8 QStep2QpRound(mfxF64 qstep)
{
    mfxU8 qp = std::lower_bound(std::begin(QSTEP), std::end(QSTEP), qstep) - std::begin(QSTEP);

    if ((qp < 51) && (qstep - QSTEP[qp] > QSTEP[qp+1] - qp))
        ++qp;

    return qp;
}
```

File: _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp [hunk 2]
```diff
@@ -1509,12 +1509,29 @@ mfxU8 LookAheadBrc2::GetQp(const BRCFrameParams& par)
...
+        qp_new++;
+    return  qp_new;
+}
 mfxU8 LookAheadBrc2::GetQpForRecode(const BRCFrameParams& par, mfxU8 curQP)
 {
-    mfxU8 qp = curQP + (mfxU8)par.NumRecode;
+    mfxU8 qp = curQP;
+    if (m_maxFrameSize < par.CodedFrameSize)
+    {
+        qp = GetNewQP(par.CodedFrameSize, m_maxFrameSize, curQP);
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp]: is it really possible that ```curQP``` > 51 here? I believe (according to the code on line 1529) that is not

File: _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp (hunk 1 above)
Comment (ogolikov) [reply _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp]: QStep2QpCeil is exist function (it is used in few places). No need to redefine it.

File: _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp (hunk 2 above)
Comment (ogolikov) [reply _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp]: Min, Max QP can be user defined.

File: _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp (hunk 1 above)
Comment (mgonchar) [reply _studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp]: But you need QStep2QpRound (according to your code). And yes, I suggest to place it at https://github.com/Intel-Media-SDK/MediaSDK/blob/0e04f5606f6cbfcbcc3c9736d40964111ba87fb7/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp#L1067

**Comment by ogolikov** [comment]
This changes doesn't touch fix LA_ICQ

**Pull request merged**